src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,54 @@
-'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.die();
+    }
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super('name');
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  unhide() {
+    this.hidden = false;
+  }
```

Comment:
[FUNCTION NAMING] - The `unhide` method is not mentioned in the task description. The `hide` method should just hide the herbivore, not toggle the `hidden` property. Consider removing the `unhide` method.